src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs

```diff
@@ -123,40 +166,91 @@ private async Task ProcessQueueAsync()
         {
             // Keep track of solution state modifications made by LSP requests
             Solution? lastMutatedSolution = null;
-            var queueToken = _cancelSource.Token;
 
-            while (!queueToken.IsCancellationRequested)
+            try
             {
-                var work = await _queue.DequeueAsync().ConfigureAwait(false);
+                while (!_cancelSource.IsCancellationRequested)
+                {
+                    var work = await _queue.DequeueAsync().ConfigureAwait(false);
```

Comment:
Potentially it means the `while (!_cancelSource.IsCancellationRequested)` condition isn't necessary now since it could get cancelled here.

Reply:
Good point about the exception, I've added a catch. Going to leave the condition on the while loop though, because if we cancel the token I think its still easier to let the DrainQueue method handle things than having this loop run through as well.